geth/jail/internal/loop/loop_test.go

```diff
@@ -0,0 +1,72 @@
+package loop
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	gomock "github.com/golang/mock/gomock"
+
+	"github.com/status-im/status-go/geth/jail/internal/vm"
+	"github.com/stretchr/testify/suite"
+)
+
+func TestLoopSuite(t *testing.T) {
+	suite.Run(t, new(LoopSuite))
+}
+
+type LoopSuite struct {
+	suite.Suite
+
+	loop   *Loop
+	cancel context.CancelFunc
+
+	mockTask     *MockTask
+	loopMockCtrl *gomock.Controller
+}
+
+func (s *LoopSuite) SetupTest() {
+	s.loopMockCtrl = gomock.NewController(s.T())
+	s.mockTask = NewMockTask(s.loopMockCtrl)
+
+	vm := vm.New()
+	s.loop = New(vm)
+
+	ctx, cancel := context.WithCancel(context.Background())
+	s.cancel = cancel
+	go s.loop.Run(ctx)
+}
+
+func (s *LoopSuite) TearDownTest() {
+	s.loopMockCtrl.Finish()
+}
+
+func (s *LoopSuite) TestAddAndReady() {
+	s.mockTask.EXPECT().SetID(int64(1)).Times(1)
+	s.mockTask.EXPECT().GetID().Times(2)
+	s.mockTask.EXPECT().Execute(s.loop.vm, s.loop).Times(1)
```

Comment:
Again, we have such checks in the legacy testing code, so it's not a good source to learn from.

Generally, tests should test "what", not "how".

There is no value in testing "how many times GetID is called". The only outcome of this test is that if the internal implementation will change (GetID is called 3 times, for example), developer is forced to update tests as well.

I would safely remove those `EXPECT` lines completely, eliminating the whole need for using `gomock` a bonus.